SawtoothSigning/Signer.swift

```diff
@@ -16,11 +16,20 @@
 //  limitations under the License.
 //
 
-/// Convenience class that wraps the PrivateKey and Context
+/// Convenience class that wraps the PrivateKey and Context.
 public class Signer {
     var context: Context
     var privateKey: PrivateKey
 
+    /**
+        Constructs a new signer.
+
+         - Parameters:
+            - context: A signing context.
+            - privateKey: A private key based on the given context.
+
+         - Returns: Signer that wraps the given context and private key.
+    */
     public init(context: Context, privateKey: PrivateKey) {
```

Comment:
The `sign` and `getPublicKey` methods throw errors. Should the doc strings have information about the errors they throw?

Reply:
Ehh that's a good question. My opinion is that we can't provide any info about the error types since Signer calls methods on the Context and PrivateKey protocols, which in turn don't specifically define the error types.